### ISSUE TITLE
[aws_wafv2_web_acl] Web ACL update-in-place fails

### ISSUE THREAD

**Issue by Tailzip** [terraform-providers/terraform-provider-aws#14035]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform version : `0.12.20`
AWS provider version : `2.68.0`

### Affected Resource(s)

* aws_wafv2_web_acl

### Terraform Configuration Files

```hcl
resource "aws_wafv2_web_acl" "waf_web_acl" {
  name        = "delete-me-demo"
  description = "to be deleted..."
  scope       = "REGIONAL"

  default_action {
    allow {}
  }

  rule {
    name     = "rate-limit"
    priority = 0

    action {
      block {}
    }

    statement {
      rate_based_statement {
        limit              = 1000
        aggregate_key_type = "IP"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = false
      metric_name                = "rate-limit-delete-me-demo-metric-name"
      sampled_requests_enabled   = false
    }
  }

  tags = {
    foo = "bar"
  }

  visibility_config {
    cloudwatch_metrics_enabled = false
    metric_name                = "delete-me-demo-metric-name"
    sampled_requests_enabled   = false
  }
}
```

### Debug Output

https://gist.github.com/Tailzip/23feef7ef2d6c325da1812e631f62a9c

### Expected Behavior

Web ACL is updated-in-place without error.

### Actual Behavior

Web ACL isn't updated-in-place. `terraform apply` fails with the following error message :
```
Error: Error updating WAFv2 WebACL: WAFInvalidParameterException: Error reason: Your statement has multiple values set for a field that requires exactly one value., field: STATEMENT, parameter: Statement
{
  RespMetadata: {
    StatusCode: 400,
    RequestID: "11e05729-fdb9-44e2-bbdc-8677eb7302c8"
  },
  Field: "STATEMENT",
  Message_: "Error reason: Your statement has multiple values set for a field that requires exactly one value., field: STATEMENT, parameter: Statement",
  Parameter: "Statement",
  Reason: "Your statement has multiple values set for a field that requires exactly one value."
}
```
In the debug log attached, you can clearly see that the JSON body of the POST request to update the web ACL is **missing data under the `Statement` key** :
```json
{
  "DefaultAction": {
    "Block": {}
  },
  ...
  "Rules": [
    {
      "Action": {
        "Block": {}
      },
      "Name": "rate-limit",
      "Priority": 0,
      "Statement": {},
      "VisibilityConfig": {
        "CloudWatchMetricsEnabled": false,
        "MetricName": "rate-limit-delete-me-demo-metric-name",
        "SampledRequestsEnabled": false
      }
    }
  ],
  ...
}
```

### Steps to Reproduce

1. `terraform apply`
2. Update `default_action` block content from `allow {}` to `block {}`
3. `terraform plan` shows :
```sh
# aws_wafv2_web_acl.waf_web_acl will be updated in-place
~ resource "aws_wafv2_web_acl" "waf_web_acl" {
  ...
  ~ default_action {
    - allow {}

    + block {}
  }
  ...
}
```
4. `terraform apply` fails...

**Comment by ewbankkit** [comment]
Similar:
* https://github.com/terraform-providers/terraform-provider-aws/issues/14029

**Comment by whizz0** [comment]
I'm getting the same error while TF plan shows update only in AWS tags, nothing else.
Using aws provider v. 2.70.0

**Comment by anGie44** [comment]
A fix has been merged and will release with `v3.2.0` of the Terraform AWS Provider, expected out later today.
